### PR TITLE
Add Bazel central registry files

### PR DESCRIPTION
**Pull request by luispadron**
Depends on #776.

This adds the required template files for https://github.com/bazel-contrib/publish-to-bcr an action which will publish to the Bazel central registry when new releases are tagged.

We'll need someone with repository admin access to approve the action configuration I requested.

### CODE REVIEW

File: .bcr/metadata.template.json
Comment (luispadron) [inline .bcr/metadata.template.json:9]: @jszumski @thiagohmcruz @mattrobmattrob @jerrymarino @justinseanmartin please provide the information needed here, you're the first that come to mind as active maintainers here but if we have more people we'd like listed please provide those here too.

I dont have access to the repo settings but maybe listing all users with approval / merge access here makes sense.

File: .bcr/config.yml
Comment (luispadron) [inline .bcr/config.yml:3]: For now we'll need to mark a single person as releaser. See an example PR for what this does:

- PR Example: https://github.com/bazelbuild/bazel-central-registry/pull/1010
- Information on this config: https://github.com/bazel-contrib/publish-to-bcr#a-note-on-release-automation

In the future we could create a bot account to handle this.

File: .bcr/presubmit.yml
Comment (luispadron) [inline .bcr/presubmit.yml:12]: Our release archive doesnt contain the `tests` directory so we just care about building here. This is similar to other repos that maintain minimal release archives. 

Relying on rules_ios CI is good enough IMO and ensuring the release stuff builds is probably good enough as a starting point for this check.

File: .bcr/config.yml
Comment (jerrymarino) [reply .bcr/config.yml:3]: Perhaps for rules_ios the release author is a shared google group email?

File: .bcr/metadata.template.json
Comment (jerrymarino) [reply .bcr/metadata.template.json:9]: Is this 100% necessary to add this field? We may have to be mindful of what emails to add to the repo - will try to start a thread offline

File: .bcr/config.yml
Comment (luispadron) [reply .bcr/config.yml:3]: Makes sense, though it needs to be a GitHub account with a fork of the BCR since that's how it makes the PR to release. If you want to make that account that would be great and I can get this PR updated

File: .bcr/metadata.template.json
Comment (luispadron) [reply .bcr/metadata.template.json:9]: Seems to be required from the template/other repos I'm seeing. We could make this that bot account though if we make it

Comment (jerrymarino) [reply .bcr/metadata.template.json:9]: Yeah ideally we can setup a google group / account for it

Comment (mattrobmattrob) [reply .bcr/metadata.template.json:9]: Would be interesting to try excluding the `maintainers` but just guessing that's a possibility based on https://github.com/bazelbuild/bazel-central-registry/blob/main/docs/README.md#validations

**Comment by luispadron** [comment]
Are we okay with merging this with me as the releaser and re-visiting this to add more info / use a shared account when that's possible.

This essentially just marks me as the maintainer for the BCR part of rules_ios which I'm okay with. I'd just like to unblock this if possible

**Comment by luispadron** [comment]
Updated PR to:

- Remove config.yml, this means no one will get tagged since it'll tag the bot like [here](https://github.com/bazelbuild/bazel-central-registry/pull/990).
- Set the shared Google workspace email as the sole maintainer.

We can attempt with this setup and followup as needed.

**Review by justinseanmartin — APPROVED** [review]

**Pull request merged**
